src/redux/actions/cellSets/runCellSetsClustering.js

```diff
@@ -36,7 +36,20 @@ const runCellSetsClustering = (experimentId, resolution) => async (dispatch, get
   });
 
   try {
-    await sendWork(experimentId, REQUEST_TIMEOUT, body, status);
+    await fetchWork(experimentId, body, status, {
+      timeout: REQUEST_TIMEOUT,
+      eventCallback: (err, res) => {
+        if (err) {
+          throw err;
+        }
+
+        const louvainSets = JSON.parse(res.results[0].body);
+        const newCellSets = [
+          louvainSets,
+        ];
+        dispatch(updateCellSetsClustering(experimentId, newCellSets));
+      },
```

Comment:
Why is this `eventCallback` in the ui that triggered this request? We want this logic to also be present in the other ui's that are currently browsing this experiment, right? (otherwise we could run into problems where one ui has an older louvain clusters version and it renames a cluster or something and so it overwrites the new louvain clusters with its old version).

Reply:
Yes, in fact, we are not meant to have this here at all. I will comment it out and instead add an explanation for how to use it as an example in case someone were to create something that takes an `eventCallback` in the future.

For now, we won't update clustering information for a user if another user updated the clusters for the same experiment. This is because that would actually cause inconsistent behavior, as we do not do this for any other work type either (e.g. for embeddings).